rest/data/host_create.go

```diff
@@ -157,6 +162,56 @@ func CreateHostsFromTask(ctx context.Context, env evergreen.Environment, t *task
 	return catcher.Resolve()
 }
 
+// GetHostCreateDistro returns the distro based on the name and provider.
+// If the provider is Docker, passing in the distro is required, and the
+// distro must be a Docker distro. If the provider is EC2, the distro
+// name is optional.
+func GetHostCreateDistro(ctx context.Context, createHost apimodels.CreateHost) (*distro.Distro, error) {
+	var d *distro.Distro
+	var err error
+	isDockerProvider := evergreen.IsDockerProvider(createHost.CloudProvider)
+	if isDockerProvider {
+		d, err = distro.FindOneId(ctx, createHost.Distro)
+		if err != nil {
+			return nil, errors.Wrapf(err, "finding distro '%s'", createHost.Distro)
+		}
+		if d == nil {
+			return nil, errors.Errorf("distro '%s' not found", createHost.Distro)
+		}
+		if !evergreen.IsDockerProvider(d.Provider) {
+			return nil, errors.Errorf("distro '%s' provider must support Docker but actual provider is '%s'", d.Id, d.Provider)
+		}
+	} else {
+		d = &distro.Distro{}
```

Comment:
nit: this if-else + defaulting of distro reads a little awkwardly since you have `d` starting out as a nil pointer, then for EC2, it gets initialized as an empty distro, then gets further overridden to an actual distro if the user specifies a distro name. It might read a little nicer if, rather than setting it 3 times, we either set the initial `d` value to `&distro.Distro{}` unconditionally, or move this defaulting into an `else` following `if createHost.Distro != ""`.